src/main/java/bio/terra/service/snapshot/flight/create/SnapshotCreateFlight.java

```diff
@@ -113,6 +113,8 @@ public SnapshotCreateFlight(FlightMap inputParameters, Object applicationContext
     AuthenticatedUserRequest userReq =
         inputParameters.get(JobMapKeys.AUTH_USER_INFO.getKeyName(), AuthenticatedUserRequest.class);
 
+    UUID snapshotId = inputParameters.get(JobMapKeys.SNAPSHOT_ID.getKeyName(), UUID.class);
```

Comment:
This is a breaking change with previous flights. To detect the breakage earlier, I would wrap this in a `Objects.requireNonNull()` so it fails before stairway tries to run a step. You can include a more specific error message for this case too, although that would only be useful for a short time after it's deployed, and then all new flights would have the input parameter set.

Reply:
> Am I understanding correctly that this could only pose an issue during TDR deployments where snapshot creation flights were previously submitted, still running during deployment, and picked up by the new Stairway pod?

Yes, exactly